FirebaseCoreSwift/Sources/FirebaseAppInternal.swift

```diff
@@ -0,0 +1,97 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import FirebaseCore
+import ObjectiveC
+
+extension FirebaseApp {
+  /// A flag indicating if this is the default app (has the default app name). Redirects to the
+  /// Obj-C property of the same name.
+  var isDefaultApp: Bool {
+    // First two arguments are the class (FirebaseApp) and the selector name.
+    typealias VoidBoolFunc = @convention(c) (AnyObject, Selector) -> Bool
+    let sel = NSSelectorFromString("isDefaultApp")
+    let isDefaultAppFunc = generatePrivateFunc(sel, from: self, type: VoidBoolFunc.self)
+    return isDefaultAppFunc(self, sel)
+  }
+
+  /// Checks if the default app is configured without trying to configure it. Redirects to the Obj-C
+  /// method of the same name.
+  static var isDefaultAppConfigured: Bool {
+    // First two arguments are the class (FirebaseApp) and the selector name.
+    typealias VoidBoolFunc = @convention(c) (AnyClass, Selector) -> Bool
+    let sel = NSSelectorFromString("isDefaultAppConfigured")
+    let isDefaultAppConfigured = generatePrivateClassFunc(sel, from: self, type: VoidBoolFunc.self)
+    return isDefaultAppConfigured(self, sel)
+  }
+
+  /// Registers a given third-party library with the given version number to be reported for
+  /// analytics. Redirects to the Obj-C method of the same name.
```

Comment:
`analytics` is overloaded. Is there a better term?

Reply:
I'll come up with something - I copied straight from the old header.

Reply:
Actually, removed the function since we don't need it right now.